lib/spack/spack/package.py

```diff
@@ -868,7 +868,8 @@ def _resource_stage(self, resource):
     def do_install(self,
                    keep_prefix=False,
                    keep_stage=False,
-                   ignore_deps=False,
+                   install_deps=True,
+                   install_self=True,
```

Comment:
Sorry if I am late at being picky here, but I find the API of `do_install` growing quite unintuitive. Now we may call a `do_install` that does no install at all, while the semantic needed for this PR could have been obtained directly in the `install` command traversing the DAG with or without root and calling an unmodified version of `do_install` from there.